API/Startup.cs

```diff
@@ -226,52 +232,61 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
-                            } else
-                            {
-                                newUser.Role = registeredUserRole;
-                                userService.Add(newUser);
-                            }
-                            await dbContext.SaveChangesAsync().ConfigureAwait(false);
-                        }
+                                    identityId = context.User.GetIdentityId(context);
+                                } catch(UnauthorizedAccessException e)
+                                {
+                                    Log.Logger.Error(e, "User is not authorized.");
+                                    await next();
+                                }
+                                if(await userService.GetUserByIdentityIdAsync(identityId)
+                                                    .ConfigureAwait(false) ==
+                                   null)
+                                {
+                                    IRoleService roleService = context.RequestServices.GetService<IRoleService>();
+                                    Role registeredUserRole =
+                                        (await roleService.GetAll()).FirstOrDefault(
+                                            i => i.Name == nameof(Defaults.Roles.RegisteredUser));
 
-                        await next().ConfigureAwait(false);
-                    });
-                });
+                                    User newUser = context.GetUserInformation(Config);
+                                    if(newUser == null)
+                                    {
+                                        // Then it probably belongs swagger so we set the username as developer.
+                                        newUser = new User()
+                                                  {
+                                                      Name = "Developer",
+                                                      Email = "Developer@DEX.com",
+                                                      IdentityId = identityId,
+                                                      Role = registeredUserRole
+                                                  };
+                                        userService.Add(newUser);
+                                    } else
+                                    {
+                                        newUser.Role = registeredUserRole;
+                                        userService.Add(newUser);
+                                    }
+                                    await dbContext.SaveChangesAsync()
+                                                   .ConfigureAwait(false);
+                                }
+
+                                await next()
+                                    .ConfigureAwait(false);
+                            });
+                        });
 
             app.UseEndpoints(endpoints => endpoints.MapDefaultControllerRoute());
 
```

Comment:
Ik weet niet wat hier precies in veranderd is/wat het doet

Reply:
Just compared with older version of this file. Only formatting is changed.